src/cpp/src/block_manager.hpp

```diff
@@ -782,6 +782,8 @@ class BlockManager {
      * @param seq_id Identifier of the sequence to free.
      */
     void free_sequence(size_t seq_id) {
+        std::lock_guard<std::mutex> lock(m_cached_blocks_map_mutex);
```

Comment:
what's about other methods which modify `m_block_table` ?

Reply:
Locally I didn't reproduce issue with them, but I think we can add mutex to them too to be safe.

Reply:
Done.